From 719b2d03b9eb63047e681644514a0e42d232a893 Mon Sep 17 00:00:00 2001
From: Patrick Combes <patrick.combes@intel.com>
Date: Tue, 30 Apr 2013 09:16:48 +0200
Subject: [PORT FROM MAIN][BT] Fix possible uninitialized variable

BZ: 104381

KW reports possible unitialized variable with usage of 'p_name'
'tmp_path' should be preferred over 'p_name' to avoid this unitialized
pointer condition.

Change-Id: Ib6b0cd0f3a06f6d250cd9b8302fcd9f51885b45d
Orig-Change-Id: Ie40c86dc9d430635e9f4690fedc75fc59934b8a7
Feature: BT
Category: Improvement
Purpose: AOSP+
Origin: Internal
Upstream-candidate: yes
Signed-off-by: Patrick Combes <patrick.combes@intel.com>
Reviewed-on: http://android.intel.com:8080/105503
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libbt/src/hardware.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libbt/src/hardware.c b/libbt/src/hardware.c
index 353205f..d9cc953 100644
--- a/libbt/src/hardware.c
+++ b/libbt/src/hardware.c
@@ -744,7 +744,7 @@ check_local_name:
                     {
                         if ((hw_cfg_cb.fw_fd = open(tmp_path, O_RDONLY)) == -1)
                         {
-                            ALOGE("vendor lib preload failed to open [%s]", p_name);
+                            ALOGE("vendor lib preload failed to open [%s]", tmp_path);
                         }
                         else
                         {
-- 
1.7.4.1

